### DISCUSSION TITLE
Not able to delete Completed jobs

### DISCUSSION THREAD

**Discussion by arrabito** [DIRACGrid/DIRAC#5686] [Q&A] [answered]
I'm trying to delete some completed jobs, but it seems that it's not possible.

Is it a wanted feature? If yes, how do you suggest to proceed?

Thank you.

**Comment by arrabito** [comment]
I've checked that indeed this is what is implemented in the StateMachine:

https://github.com/DIRACGrid/DIRAC/blob/8d242fc992f5853974c09b05135a956abb0e8bc0/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py#L72

The status from Completed can only go into Done or Failed.

The problem in my case is that the Request associated with the Job is Failed, but the jobs stays Completed forever...

Shouldn't go into Failed as well?

**Comment by chaen** [comment, marked as answer]
Well, it is not clear what to do with a Job if the request fails. There are cases where you just don't care (for example, imagine a failover request which is just for accounting). 
So the policy is to do nothing.
In general, a Request should never fail, so it always requires assistance :-)

**Reply by arrabito** [reply]
OK thanks, I see your point. So a related question is if there any command to set Completed jobs either to Done, either to Failed.

I found that one can do that using:

```
DIRAC.WorkloadManagementSystem.Client.JobReport
...
jobReport.setJobStatus("Failed")
```

but I don't know if this is the optimal way to do it.

Thanks.

**Reply by fstagni** [reply]
As In 7.3 there's a WMS state machine (the allowed transitions are defined in https://github.com/DIRACGrid/DIRAC/blob/rel-v7r3/src/DIRAC/WorkloadManagementSystem/Client/JobStatus.py#L72).

Anyway:
```
from DIRAC.WorkloadManagementSystem.Client.JobStateUpdateClient import JobStateUpdateClient
JobStateUpdateClient().setJobStatus(jobID, status="", minorStatus="", force=True)
```
would do it.

**Comment by arrabito** [comment]
OK Thank you.
